src/sys/private/sockets_posix.nim

```diff
@@ -164,14 +164,7 @@ proc handleAsyncConnectResult(fd: SocketFD) {.raises: [OSError].} =
   if error != 0:
     raise newOSError(error, $Error.Connect)
 
-template tcpConnect() {.dirty.} =
-  let addressFamily =
-    when endpoint is IP4Endpoint:
-      AF_INET
-    elif endpoint is IP6Endpoint:
-      AF_INET6
-  let sock = makeSocket(addressFamily, SOCK_STREAM, IPPROTO_TCP)
-
+template socketConnect() {.dirty.} =
```

Comment:
Can you make this take parameters normally? Aside from the implementations I would prefer if none of the templates are `{.dirty.}`